Sage/carbon
# fix(select): render select list in DOM order rather than in a portal

The select list is no longer rendered inside a portal, but after the select input in DOM order. This ensures that screenreaders will move directly through the list options when focus is on the input and the user steps through the following content.

BREAKING CHANGE: The `disablePortal` prop has been removed from all 3 Select variants. It is no longer needed as the component now always behaves as it did before with this prop added.

fix #5171

### Proposed behaviour



When the Select input is focused with the select list open, and a screenreader is told to move to the "next" element (Control+Option+RightArrow is the default on VoiceOver), it should read through the option list, no matter what content may occur below/after the select.

### Current behaviour



Using the above-mentioned key combination moves to any content after the select, only coming to the select list and its options after reaching the bottom of the page.

### Checklist




#### QA


### Additional context



This fix should also remove the Axe error about content not being contained in landmarks (see #4452). To see this, you need to run Axe with "best practices" turned ON. Note that on Storybook Select examples this will still bring up errors as we do not put the select inside an element with a landmark role, but if a `Select` is rendered inside eg a `main` HTML element, previously this error would still come up when the select list was open, and now it will not.

See @tempertemper 's comments on the just-linked issue, which suggested the fix that has been implemented in this PR.

### Testing instructions



Can be tested with screenreaders on the storybook docs for each of the 3 selects - test on the "docs" tab rather than "canvas" to ensure there is content between the Select being tested and the bottom of the page.

Test with NVDA on Windows as well as VoiceOver on Mac.

Test Select (all 3 varieties) for any regressions.
